src/equations/compressible_navier_stokes_1d.jl

```diff
@@ -215,10 +215,10 @@ end
     return cons2prim(entropy2cons(u_transformed, equations), equations)
 end
 
-# Takes the solution values `u` and gradient of the entropy variables (w_2, w_3, w_4) and
-# reverse engineers the gradients to be terms of the primitive variables (v1, v2, T).
+# Takes the solution values `u` and gradient of the entropy variables w and
+# reverse engineers the gradients to be terms of the primitive variables u_prim = (rho, v1, v2, T).
```

Comment:
Same here. I would like to avoid unnecessary gradients as it wastes computation time and memory. Although, I understand that having a gradient vector of variables that is one less than the conservative variable vector could be an annoying restructuring of the existing code.